lib/config/rules/style.js

```diff
@@ -52,7 +52,7 @@ module.exports = {
   // Require or disallow a space before function parenthesis
   'space-before-function-paren': require('./rule-helpers/space-before-function-paren-rule'),
   // Checks variable names for various errors.
-  'variable-name': [true, 'ban-keywords', 'check-format', 'allow-leading-underscore'],
+  'variable-name': [true, 'ban-keywords', 'check-format', 'allow-pascal-case'],
```

Comment:
Cool, this makes sense to me. Might make sense to only include this as part of the React config, though (non-React projects probably shouldn't have pascal case variables).

Reply:
@ismail-syed 

To prevent "xxx is declared but never used", `allow-leading-underscore` was pretty useful when dealing with functions like:

```js
  private handleOnBeforeChange(_editor, _data, value) {
    this.setState({value});
  }
```

Instead we have to do something like this:

```js
  private handleOnBeforeChange(...args) {
    const [value] = args;
    this.setState({value});
  }
```

Unless there's a workaround I'm missing?

Reply:
IIRC, I think we prefer the second approach more. 
@lemonmade, is that correct?

Reply:
It allows us to pass in more function args without having to update all the function calls.

Reply:
Actually, for some reason doing this:

```
  private handleOnBeforeChange(...args) {
    const [value] = args;
    this.setState({value});
  }
```

Made some stuff fail. Not sure why, maybe you need this:

```
  private handleOnBeforeChange(...args) {
    const [value, ...rest] = args;
    this.setState({value});
  }
```

Which then leads to the unused `rest` var 🗡

Reply:
I am fine with the leading underscore, personally.

Reply:
@kaelig feel free to submit a PR.

Reply:
Thank you for the feedback! See https://github.com/Shopify/tslint-config-shopify/pull/80